### PR TITLE
console: print promise details

### PR DESCRIPTION
**Pull request by mhvsa**
Related with https://github.com/denoland/deno/issues/3165

Current:
```js
Promise {}
Promise {}
Promise {}
```

Target:
```js
Promise { <pending> }
Promise { 2 }
Promise { <rejected> 'rejection reason' }
```

### CODE REVIEW

File: cli/js/globals.ts [hunk 1]
```diff
@@ -102,6 +102,8 @@ declare global {
 
     formatError: (e: Error) => string;
 
+    getPromiseDetails(promise: Promise<unknown>) : [number, unknown];
```
Comment (kitsonk) [inline cli/js/globals.ts]: This should be typed a lot more specifically.  If the second argument of the tuple is the value of the promise, then:

```suggestion
    getPromiseDetails<T>(promise: Promise<T>) : [number, T];
```

Also provide some JSDoc please, so this is easily documented.

File: core/bindings.rs [hunk 2]
```diff
@@ -761,3 +775,57 @@ pub fn module_resolve_callback<'s>(
...
+    v8::PromiseState::Fulfilled => {
+      res.set(
+        context,
+        v8::Integer::new(scope, 0).into(),
+        v8::Integer::new(scope, 1).into(),
+      );
+      res.set(
+        context,
+        v8::Integer::new(scope, 0).into(),
+        promise.result(scope).into(),
```
Comment (kitsonk) [inline core/bindings.rs]: run `cargo clippy` to pick up these linting errors.

File: core/bindings.rs [hunk 3]
```diff
@@ -761,3 +775,57 @@ pub fn module_resolve_callback<'s>(
...
   None
 }
+
+fn get_promise_details(
+  scope: v8::FunctionCallbackScope,
+  args: v8::FunctionCallbackArguments,
+  mut rv: v8::ReturnValue,
+) {
+  if !args.get(0).is_promise() {
+    return;
```
Comment (kitsonk) [inline core/bindings.rs]: I'm not sure how this code is going to behave at runtime with this.  It shouldn't silently fail.  We should effectively raise a JS error with appropriate information, like a `TypeError`.

Comment (mhvsa) [reply core/bindings.rs]: It returned `undefined` in that way; however I like the idea of throwing `TypeError` more.

**Comment by mhvsa** [comment]
Review Ready
@ry @bartlomieju @kitsonk

**Pull request merged**